tests/compiler/std/typedarray.optimized.wat

```diff
@@ -5512,36 +5512,44 @@
   (local $3 i32)
   (local $4 i32)
   (local $5 i32)
-  (local $6 i32)
   local.get $0
   call $~lib/rt/pure/__retain
   local.tee $1
   i32.load offset=8
-  local.set $2
+  local.set $3
   local.get $1
   i32.load offset=4
-  local.set $3
-  local.get $2
+  local.set $4
+  local.get $3
   call $~lib/typedarray/Int8Array#constructor
-  local.tee $4
-  call $~lib/rt/pure/__retain
-  local.tee $5
+  local.tee $0
+  local.tee $2
+  if
+   local.get $2
+   call $~lib/rt/pure/__retain
+   drop
+   i32.const 0
+   call $~lib/rt/pure/__release
```

Comment:
@dcodeIO Hmm, this shouldn't change code at all. But currently produce this extra retain/release block